plenty_api/api.py

```diff
@@ -824,7 +825,69 @@ def plenty_api_set_image_availability(self,
                                              path=path,
                                              data=data)
 
-        if not response:
-            return False
+        return response
 
-        return True
+# PUT REQUESTS
+
+    def book_incoming_items(self,
+                            article_item_id: int,
+                            variation_id: int,
+                            quantity: float,
+                            warehouseId: int,
+                            batch: str = None,
+                            bestBeforeDate: str = None) -> bool:
+        data = {
+            "warehouseId": warehouseId,
+            "deliveredAt": datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ"),
+            "currency": "EUR",
+            "quantity": quantity,
+            "reasonId": 181,
+        }
+
+        if batch:
+            data.update(batch=batch)
+        if bestBeforeDate:
+            data.update(bestBeforeDate=bestBeforeDate)
```

Comment:
I believe the `bestBeforeDate` parameter should be parsed first by the `utils.parse_date` function, which makes that a) the string contains a date and b) that it is in the W3C format.
something along the lines of:
```
if bestBeforeDate:
    w3c_date = utils.parse_date(date=bestBeforeDate)
    if w3c_date:
        data.update(bestBeforeDate=w3c_date)
```

Reply:
Good call, thank you.